algorithms/unit_tests/CMakeLists.txt

```diff
@@ -6,34 +6,6 @@ KOKKOS_INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/../src )
 KOKKOS_INCLUDE_DIRECTORIES(${KOKKOS_SOURCE_DIR}/core/unit_test/category_files)
 
 
-SET(GTEST_SOURCE_DIR ${${PARENT_PACKAGE_NAME}_SOURCE_DIR}/tpls/gtest)
-KOKKOS_INCLUDE_DIRECTORIES(${GTEST_SOURCE_DIR})
-
-# mfh 03 Nov 2017: The gtest library used here must have a different
-# name than that of the gtest library built in KokkosCore.  We can't
-# just refer to the library in KokkosCore's tests, because it's
-# possible to build only (e.g.,) KokkosAlgorithms tests, without
-# building KokkosCore tests.
-
-
-KOKKOS_ADD_TEST_LIBRARY(
-  kokkosalgorithms_gtest
-  HEADERS ${GTEST_SOURCE_DIR}/gtest/gtest.h
-  SOURCES ${GTEST_SOURCE_DIR}/gtest/gtest-all.cc
-)
-
-# avoid deprecation warnings from MSVC
-TARGET_COMPILE_DEFINITIONS(kokkosalgorithms_gtest PUBLIC GTEST_HAS_TR1_TUPLE=0 GTEST_HAS_PTHREAD=0)
-
-IF((NOT (Kokkos_ENABLE_CUDA AND WIN32)) AND (NOT ("${KOKKOS_CXX_COMPILER_ID}" STREQUAL "Fujitsu")))
-  TARGET_COMPILE_FEATURES(kokkosalgorithms_gtest PUBLIC cxx_std_14)
-ENDIF()
-
-# Suppress clang-tidy diagnostics on code that we do not have control over
-IF(CMAKE_CXX_CLANG_TIDY)
-  SET_TARGET_PROPERTIES(kokkosalgorithms_gtest PROPERTIES CXX_CLANG_TIDY "")
-ENDIF()
-
```

Comment:
Don't we need that for `Trilinos` testing when building separate libraries?

Reply:
I could have been overzealous - I haven't tested this with `Trilinos`.

Reply:
For reference purposes, PR #1212 introduced the changes to add `kokkosalgorithms_gtest` and  contains additional context for it's need.
When building Trilinos with tests enabled generically (i.e. `-DKokkos_ENABLE_TESTS=ON`) the algorithm tests built and tested fine.
Selectively enabling sub-package tests (e.g. `-DKokkosAlgorithms_ENABLE_TESTS=ON -DKokkosCore_ENABLE_TESTS=OFF`) led to a configuration error like that in #1212. I tested the develop branch which also has the configuration error, so the changes in this PR did not introduce the error though I suspect that we will need an updated version of this block of CMake code if we are to resolve the issue of enabling sub-package testing

Reply:
@crtrott do we really to support enabling tests in the algorithm package but not in core?

Reply:
Confirmed at Kokkos Developer Meeting: we don't need to support that unless someone complains.